fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt

```diff
@@ -166,3 +177,50 @@ fun writeAdditionalTestApps(strs: List<String>, flankGradleExtension: FlankGradl
     }
   )
 }
+
+val Project.isAndroidAppModule
+  get() = plugins.hasPlugin("com.android.application")
+val Project.isAndroidLibraryModule
+  get() = plugins.hasPlugin("com.android.library")
+
+// returns false if the module explicitly disabled testing or if it simply had no tests
+val Project.hasAndroidTest: Boolean
+  get() {
+    val fulladleModuleExtension = extensions.findByType(FulladleModuleExtension::class.java) ?: return false
+    if ((isAndroidLibraryModule || isAndroidAppModule) && !fulladleModuleExtension.enabled.get()) {
+      return false
+    }
+
+    val testedExtension = extensions.getByType<TestedExtension>()
+    var testsFound = true
+    testedExtension.testVariants.configureEach testVariant@{
+      if (!file("$projectDir/src/androidTest").exists()) {
```

Comment:
this touches a file, so we should make sure this is method is not called during the Gradle configuration phase.

Reply:
This was already present in the configuration block. I just moved it to another method called from the configuration block. I'm not sure if checking file presence modifies/touches it.

Reply:
Makes sense. This isn't related then.